types/parser/parse.go

```diff
@@ -0,0 +1,111 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/antlr4-go/antlr/v4"
+	substraitgo "github.com/substrait-io/substrait-go"
+	"github.com/substrait-io/substrait-go/types"
+	baseparser2 "github.com/substrait-io/substrait-go/types/parser/baseparser"
+)
+
+type TypeExpression struct {
+	ValueType types.FuncDefArgType
+}
+
+func (t *TypeExpression) MarshalYAML() (interface{}, error) {
```

Comment:
A test to test the linkage for marshaling into/out of YAML would likely be useful.

Reply:
added